### ISSUE TITLE
Gnome: avoid GUI subprocesses starting as alacritty windows?

### ISSUE THREAD

**Issue by sztomi** [alacritty/alacritty#4765]
### System

OS: Linux
Version: `alacritty 0.5.0-dev (0677d230)`
Linux/BSD: X11, Gnome Shell

I noticed that when I start a GUI subprocess from alacritty, even if the process detaches from the shell, its window will appear as if it was an alacritty window. 

![image](https://user-images.githubusercontent.com/240594/107013489-b721c400-679a-11eb-927f-5b11b80dc7e7.png)

Specifically, this happened with the apt version of vscode (starting as `code .` in alacritty). Note, this did not happen when I used the snap version. Is there a way to avoid this behavior? Apologies if this is not an alacritty bug, I wanted to check here first as this is the only terminal I see this behavior with (VTE-based terminals such as Gnome Terminal don't seem to exhibit this behavior).

I also saw this with bcompare, however, that command does not detach so I can understand why that window would be counted as an alacritty window.

**Comment by chrisduerr** [comment]
This sounds like a GNOME specific problem? I'm not sure what strange strategies they use to determine this, but there's likely some proprietary protocol or environment variable required to change this.

**Comment by sztomi** [comment]
@chrisduerr yeah, I'm not sure either. I was wondering if the alacritty window could be created with some flag to indicate that it's a terminal won't have sub-windows spawned? not sure.

**Comment by mqudsi** [comment]
Gut instinct w/ my shell developer hat on: can you please post a screenshot/the output of `ps`/`htop`/whatever with tree mode enabled and showing both the pid and pgrp?

**Comment by sztomi** [comment]
@mqudsi This is the apt-installed code (this is where the Code window appears as an alacritty window):

![image](https://user-images.githubusercontent.com/240594/107154580-f35c4c80-6973-11eb-9e16-f07ef1fb795e.png)

And alacritty at the same time:

![image](https://user-images.githubusercontent.com/240594/107154610-1edf3700-6974-11eb-8f5c-64f328ad0bd5.png)

(note that `code` is not a child process of alacritty)

This is snap-installed code (this is where the Code window appears correctly as a separate application): 

![image](https://user-images.githubusercontent.com/240594/107154659-7382b200-6974-11eb-840c-10002512a6d7.png)

and alacritty at the same time (same instance as before):

![image](https://user-images.githubusercontent.com/240594/107154669-8d23f980-6974-11eb-812d-aaa281d95a73.png)

**Comment by Davidian1024** [comment]
I just installed Ubuntu 22.04 fresh and decided to give alacritty a try.  I'm seeing this exact behavior.

I found this bug report while searching for a way to get to stop happening.

FWIW, rxvt-unicode (urxvt) v9.30 that I installed via apt-get doesn't behave this way.  Any gui apps I run from within it get their own icons on the dock with that terminal emulator.

That said, I would also like a way to change this behavior when running gui apps from alacritty.

**Comment by Davidian1024** [comment]
So, I did a little more testing.  Partly out of curiosity and partly because I'm really liking alacritty so far.
The issue is actually worse than I realized at first.  It's not only gui apps launched directly from alacritty that get treated as subwindows of alacritty, it's also anything launched by the things that were launched by alacritty.

In my case it's Steam.  When I launch Steam under alacritty it get's treated as an additional alacritty window and dock item.  And any game I launch from Steam also gets added to the growing list of alacritty windows/dock items.

I went ahead and tried a few other terminal emulators and I'm not able to find any that cause this behavior.
gnome-terminal, rxvt-unicode, konsole, xterm.  Any gui apps I launch from these get treated as if they're their own desktop app, with their own icon on the dock.
